src/components/daydream-controls.js

```diff
@@ -31,20 +31,21 @@ module.exports.Component = registerComponent('daydream-controls', {
   // 1 - menu ( never dispatched on this layer )
   // 2 - system ( never dispatched on this layer )
   mapping: {
-    axis0: 'trackpad',
-    axis1: 'trackpad',
-    button0: 'trackpad',
-    button1: 'menu',
-    button2: 'system'
+    axes: {'trackpad': [0, 1]},
+    buttons: ['trackpad', 'menu', 'system']
   },
 
+  // Use these labels for detail on axis events such as thumbstickmoved.
+  // e.g. for thumbstickmoved detail, the first axis returned is labeled x, and the second is labeled y.
+  axisLabels: ['x', 'y', 'z', 'w'],
+
   bindMethods: function () {
     this.onModelLoaded = bind(this.onModelLoaded, this);
     this.onControllersUpdate = bind(this.onControllersUpdate, this);
     this.checkIfControllerPresent = bind(this.checkIfControllerPresent, this);
     this.removeControllersUpdateListener = bind(this.removeControllersUpdateListener, this);
-    this.onGamepadConnected = bind(this.onGamepadConnected, this);
-    this.onGamepadDisconnected = bind(this.onGamepadDisconnected, this);
+    this.onAxisMoved = bind(this.onAxisMoved, this);
+    this.onGamepadConnectionEvent = bind(this.onGamepadConnectionEvent, this);
```

Comment:
We can get rid of this since we can do `window.addEventListener('gamepaddisconnected', this.checkIfControllerPresent, false);`